packages/engine/paima-sm/src/types.ts

```diff
@@ -71,6 +71,23 @@ interface CdeDatumCardanoPoolPayload {
   pool: string | undefined;
 }
 
+interface CdeDatumCardanoProjectedNFTPayload {
+  ownerAddress: string;
+
+  actionTxId: string;
+  actionOutputIndex: number | undefined;
+
+  previousTxHash: string | undefined;
+  previousTxOutputIndex: number | undefined;
+
+  asset: string;
+  amount: number;
```

Comment:
This is wrong. Assets are of type `uint` in the CDDL spec for Cardano which can be up to 64 bits (larger than Javascript's max 53 bits)

JSON allows for infinite precision so JSON is fine, but any time you parse JSON to generate JSON in Javascript you lose precision if it's more than 53 bits.

Reply:
good point, I've fixed the types in carp, that would require new release of client lib so it is updated here too

Reply:
updated